content/spin/kv-store-api-guide.md

```diff
@@ -0,0 +1,151 @@
+
+{{ details "Why do I need a Spin interface? Why can't I just use my own external store?" "You can absolutely still use your own external store either with the Redis or Postgres APIs, or outbound HTTP. However, if you're interested in quick, non-relational local storage without any infrastructure set-up then Spin's key value store is a great option." }}
+
+## Using Key Value Store From Applications
+
+The Spin SDK surfaces the Spin key value store interface to your language. The following characteristics are true of keys and values:
+
+* Keys as large as 256 bytes (UTF-8 encoded)
+* Values as large as 1 megabyte
+* Capacity for 1024 key value tuples
+
+The set of operations is common across all SDKs:
+
+| Operation  | Parameters | Returns | Behavior |
+|------------|------------|---------|----------|
+| `open`  | name | store  | Open the store with the specified name. If `name` is the string "default", the default store is opened for the component that was granted access in the component manifest from `spin.toml`. Otherwise, `name` must refer to a store defined and configured in a [runtime configuration file](./dynamic-configuration.md#key-value-store-runtime-configuration) supplied with the application.|
+| `get` | store, key | key | Get the value associated with the specified `key` from the specified `store`. |
+| `set` | store, key, value | - | Set the `value` associated with the specified `key` in the specified `store`, overwriting any existing value. |
+| `delete` | store, key | - | Delete the tuple with the specified `key` from the specified `store`. `error::invalid-store` will be raised if `store` is not a valid handle to an open store.  No error is raised if a tuple did not previously exist for `key`.|
+| `exists` | store, key | boolean | Return whether a tuple exists for the specified `key` in the specified `store`.|
+| `get-keys` | store | list<keys> | Return a list of all the keys in the specified `store`. |
+| `close` | store | - | Close the specified `store`. |
+
+The exact detail of calling these operations from your application depends on your language:
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+Key value functions are available in the `spin_sdk::key_value` module. The function names match the operations above. For example:
+
+
+```rust
+use anyhow::Result;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    key_value::{Error, Store},
+};
+#[http_component]
+fn handle_request(req: Request) -> Result<Response> {
+    let store = Store::open_default()?;
+    store.set("mykey", "myvalyue")?;
+    let value = store.get("mykey")?;
+     Ok(http::Response::builder().status(200).body(Some(value.into()))?)
+}
+```
+
```

Comment:
For `set`, the value argument can be of any type that implements `AsRef<[u8]>`

For `get`, the return value is of type `Vec<u8>`.  If the key does not exist then Error::NoSuchKey is returned.

The `close` operation is not surfaced; it is called automatically when the store is dropped.